### PR TITLE
WIP DNM

### PR DESCRIPTION
**Pull request by ekaley**
DO NOT MERGE

### CODE REVIEW

**Comment by ekaley** [comment]
@king-jam models has the raw data from the nws site for all of RI, MA, and CT

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      - misspell
+      - nakedret
+      - prealloc
+      - scopelint
+      - stylecheck
+      - unconvert
+      - unparam
+      - whitespace
+      - wsl
+
```
Comment (king-jam) [inline .golangci.yml:55]: bug here fixed in other instances. need to add `issues:`

File: data/main.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// 	if err == nil {
+// 		w.Write(data)
+// 	} else {
+// 		w.WriteHeader(404)
+// 		w.Write([]byte("404 Something went wrong - " + http.StatusText(404)))
+// 	}
+// }
+
+func main() {
+	fs := http.FileServer(http.Dir("/home/tom/workspace/ft-alert-bot/data/source"))
```
Comment (king-jam) [inline data/main.go:28]: Can this be changed to a relative path?

File: models/models.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+// Toast has all the ingredients
+type Toast struct {
+	gorm.Model
+	Slices uint `gorm:"primary_key:true"`
+	Status string
+	Alert  string
+}
+
+var LevelZero = &Toast{Slices: 0, Status: "None", Alert: ""}
```
Comment (king-jam) [inline models/models.go:65]: are these constants or helper factory functions? These should either be funcs or a const

File: models/models.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+var LevelOne = &Toast{Slices: 1, Status: "Low", Alert: ""}
+var LevelTwo = &Toast{Slices: 2, Status: "Guarded", Alert: ""}
+var LevelThree = &Toast{Slices: 3, Status: "Elevated", Alert: ""}
+var LevelFour = &Toast{Slices: 4, Status: "High", Alert: ""}
+var LevelFive = &Toast{Slices: 5, Status: "Severe", Alert: ""}
+
+// Repository represent the gif usecases
```
Comment (king-jam) [inline models/models.go:72]: Repository interface defined in a domain models package?

File: scraper/scraper.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/king-jam/ft-alert-bot/models"
+)
+
+// placeholder for a check without a db implementation
+var timeStamp string
```
Comment (king-jam) [inline scraper/scraper.go:16]: should wrap this in the Scraper implementation

File: .gitignore [hunk 6]
```diff
@@ -10,3 +10,6 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+bin
+./bin
```
Comment (king-jam) [inline .gitignore:14]: isn't this repetitive?

File: .gitignore [hunk 7]
```diff
@@ -10,3 +10,6 @@
...
+bin/scraper
```
Comment (king-jam) [inline .gitignore:15]: ditto

File: .golangci.yml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+# options for analysis running
+run:
+
+    # which dirs to skip: issues from them won't be reported;
+    # can use regexp here: generated.*, regexp is applied on full path;
+    # default value is empty list, but default dirs are skipped independently
+    # from this option's value (see skip-dirs-use-default).
+    skip-dirs:
+      - static/libs
```
Comment (king-jam) [inline .golangci.yml:12]: might want to remove this ref...

File: data/main.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+	http.Handle("/", fs)
+
+	log.Println("Serving weather data...")
+	http.ListenAndServe(":6969", nil)
```
Comment (king-jam) [inline data/main.go:32]: ...

File: get-docker.sh [hunk 10]
```diff
@@ -0,0 +1,476 @@
+#!/bin/sh
```
Comment (king-jam) [inline get-docker.sh:1]: why is this file in the repo? If there is a very specific reason, should it be at the top level?

File: docker-compose-postgress.yml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+      - POSTGRES_USER=snow
+      - POSTGRES_PASSWORD=snow123
+      - POSTGRES_DB=snow
+#   Maps port 54320 (localhost) to port 5432 on the container. You can change the ports to fix your needs.
+    ports:
+      - "54320:5432"
+#   Set a volume some that database is not lost after shutting down the container.
+#   I used the name postgres-data but you can changed it to something else.
+    volumes:
+      - /home/tom/Desktop/postgres-data:/var/lib/postgresql/data
```
Comment (king-jam) [inline docker-compose-postgress.yml:21]: relative path

File: docker-compose-consul.yml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+version: "3"
```
Comment (king-jam) [inline docker-compose-consul.yml:1]: why??

File: main.go [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+	"log"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/king-jam/ft-alert-bot/models"
+	"github.com/king-jam/ft-alert-bot/scraper"
+	"github.com/king-jam/ft-alert-bot/store"
+)
+
+// SCRAPEINTERVAL will be set with flags
```
Comment (king-jam) [inline main.go:14]: setting up that const with flags? :smile:

**Review by king-jam — COMMENTED** [review]
I'm bored

**Pull request merged**